flytekit/clis/flyte_cli/example.config

```diff
@@ -0,0 +1,11 @@
+# This is an example of what a config file may look like.
+# Place this in ~/.flyte/config and flyte-cli will pick it up automatically
+
+[platform]
+url=flyte.company.com
+auth=true
+
+[credentials]
+discovery_endpoint=http://corp.idp.com/.well-known/oauth-authorization-server
```

Comment:
Just to clarify, this discovery endpoint should be pointed at `flyte.corp.net` or something... and that should redirect to whatever IDP corp.net is using.  Or it can return the values itself without redirecting, but the point is, you only hard-code the main flyte server.  Thinking about it, maybe we should nix this @katrogan and just assume the extension off of the platform URL.  That's probably the cleanest thing to do.

The redirect endpoint is configured in the IDP.  It's a hardcoded fully-specified URL.  This means that the callback url (including the port) that the local server that's stood up at has to match exactly what the IDP has.  This information is not returned by the metadata/discovery endpoint.  Why i'm not sure... it may be a security concern.

Ideally, the metadata endpoint would be less restrictive and allow us to return more things.  Because of the limitations of the RFC, we also need to hardcode this in for example: https://github.com/lyft/flyteexamples/pull/97/files#diff-2e41be19539d0ba6fda0e5ce814a9123R22.  This is the scope that will be requested by pyflyte when it makes the token request.  Again, like the callback url, it has to match a hardcoded string set within the IDP, otherwise it'll fail.  It doesn't really offer anything, but we have to have it.

Reply:
Re: assuming the extension - I'm happy to do this if we can safely assume this is the major convention (which I guess sounds like it is?)

Reply:
yup https://tools.ietf.org/html/rfc8414#section-3

Reply:
Hmm very interesting...well seems like you have a handle on it, so I defer to whatever you think is best though I am biased to less config wherever possible.